src/dune/action.ml

```diff
@@ -270,17 +270,17 @@ let is_useful_to_sandbox =
       loop t
     | Progn l -> List.exists l ~f:loop
     | Echo _ -> false
```

Comment:
There is something slightly odd here:

- `is_useful_to_memoize (Redirect_out (Stdout, fn, Echo "toto")) = false`
- `is_useful_to_memoise (Write_file (fn, "toto")) = true`

Both should return the same. What do you think of changing `Redirect_out ...` to `Redirect_out ... -> memoize || loop t`?

Reply:
I'm not really knowledgeable with all the actions. If `Redirect_stdout` creates a file then yes, this is definitively a correction.

Reply:
That's the case indeed.